Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java

```diff
@@ -0,0 +1,257 @@
+		assertEquals("http://www.hhs.gov/healthit/nhin#HomeCommunityId", attributeType.getAttributeId());
+		assertEquals("http://www.w3.org/2001/XMLSchema#anyURI", attributeType.getDataType());
+		assertEquals("urn:oid:1.1.1.1.1.1.1", attributeType.getAttributeValue().get(0).getContent().get(0));
+	}
+
+	@Test
+	public void testTransformAdhocQueryToCheckPolicy() {
+		String direction = "DocumentQueryIn";
+		AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
+		eventA.setDirection("inbound");
+		AdhocQueryRequestMessageType msg = new AdhocQueryRequestMessageType();
+		AdhocQueryRequest adhocQueryRequest = new AdhocQueryRequest();
+		SlotType1 slotType1 = new SlotType1();
+		slotType1.setName("$XDSDocumentEntryPatientId");
+
+		ValueListType valueListType = new ValueListType();
+		valueListType.getValue().add("1111.4444^^^&amp;26.489.22&amp;ISO");
+		slotType1.setValueList(valueListType);
+		AdhocQueryType adhocQueryType = new AdhocQueryType();
+		adhocQueryType.getSlot().add(slotType1);
+
+		adhocQueryRequest.setAdhocQuery(adhocQueryType);
+		msg.setAdhocQueryRequest(adhocQueryRequest);
+		AssertionType assertionTypeTest = new AssertionType();
+		assertionTypeTest.setMessageId(MESSAGE_ID);
+		HomeCommunityType homeCom = new HomeCommunityType();
+		homeCom.setHomeCommunityId(TEST_HC_VAL);
+		assertionTypeTest.setHomeCommunity(homeCom);
+		msg.setAssertion(assertionTypeTest);
+		eventA.setMessage(msg);
+
+		CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+		verifycheckPolicy(result, direction);
+
+	}
+
+	@Test
+	public void testTransformAdhocQueryToCheckPolicy_Outbound() {
+		String direction = "DocumentQueryOut";
+		AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
+		eventA.setDirection("outbound");
+		AdhocQueryRequestMessageType msg = new AdhocQueryRequestMessageType();
+		AdhocQueryRequest adhocQueryRequest = new AdhocQueryRequest();
+		SlotType1 slotType1 = new SlotType1();
+		slotType1.setName("$XDSDocumentEntryPatientId");
+
+		ValueListType valueListType = new ValueListType();
+		valueListType.getValue().add("1111.4444^^^&amp;26.489.22&amp;ISO");
+		slotType1.setValueList(valueListType);
+		AdhocQueryType adhocQueryType = new AdhocQueryType();
+		adhocQueryType.getSlot().add(slotType1);
+
+		adhocQueryRequest.setAdhocQuery(adhocQueryType);
+		msg.setAdhocQueryRequest(adhocQueryRequest);
+		AssertionType assertionTypeTest = new AssertionType();
+		assertionTypeTest.setMessageId(MESSAGE_ID);
+		HomeCommunityType homeCom = new HomeCommunityType();
+		homeCom.setHomeCommunityId(TEST_HC_VAL);
+		assertionTypeTest.setHomeCommunity(homeCom);
+		msg.setAssertion(assertionTypeTest);
+		eventA.setMessage(msg);
+
+		CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+		verifycheckPolicy(result, direction);
+
+	}
+
+	private void verifycheckPolicy(CheckPolicyRequestType actualCheckPolicy, String direction) {
+
+		assertNotNull(actualCheckPolicy);
```

Comment:
You have two private methods that seem to do the exact same thing.  Should be able to combine them or combine the similar functionality.